assignment7.txt

```diff
@@ -0,0 +1,16 @@
+1. For an array {1, 2, 3, 4}
+	a. ++*p will get the variable located at *p then increment it. if *p is at index 0, new array would be {2, 2, 3, 4} with *p at index 0
+	b. *p++ will increment p (move the index forward once), then get the value of it. If *p is at index 0, then the array will be {1,2,3,4} with *p at index 1
+	c. *++p is equivalent to *p++ as the prefix ++ and * evaluate right to left.
+2. No, it depends on the type of operator.
+3. Pointers are used for functions that need to actually change their arguments or for returning an array of values
+4.1 char[]
+4.2 invalid
+4.3 int 0
+4.4 int 10
+4.5 int * a
+4.6 int 12
+4.7 int ** 
+4.8 char *
+4.9 invalid
```

Comment:
Functions in C are actually just pointers to a spot in the program where some code exists. Just like you've been creating pointers to structs, strings, and arrays, you can point a pointer at a function too. The main use for this is to pass "callbacks" to other functions, or to simulate classes and objects. In this exercise we'll do some callbacks, and in the next one we'll make a simple object system.
The format of a function pointer goes like this:

```int (*POINTER_NAME)(int a, int b)```

```So the answer is actually int(*)(int, char**). All you needed to do was evaluate the data types of the &main. Since & returns the memory address, main's data type would change from int to int*. Since argc and argv are just the parameters, they would remain the same data types. So the final answer is int(*)(int, char**).```